From 93d712c1536d702201a06f77417875ef2ae33300 Mon Sep 17 00:00:00 2001
From: yaravapa <yasoda.aravapalli@intel.com>
Date: Mon, 25 Jan 2021 10:28:15 +0530
Subject: [PATCH] Use clang toolchain from prebuilts instead of gcc

This is required to enable setup path for Android 11
Tracked-On: OAM-95316
Signed-off-by: yaravapa <yasoda.aravapalli@intel.com>
---
 Makefile               | 34 +++++++++++++++++-----------------
 kernel/gen_kheaders.sh | 11 +++++++++--
 2 files changed, 26 insertions(+), 19 deletions(-)

diff --git a/Makefile b/Makefile
index cbfbf98c3fa7..6aedf6f78d4b 100644
--- a/Makefile
+++ b/Makefile
@@ -394,13 +394,13 @@ HOST_LFS_CFLAGS := $(shell getconf LFS_CFLAGS 2>/dev/null)
 HOST_LFS_LDFLAGS := $(shell getconf LFS_LDFLAGS 2>/dev/null)
 HOST_LFS_LIBS := $(shell getconf LFS_LIBS 2>/dev/null)
 
-ifneq ($(LLVM),)
+#ifneq ($(LLVM),)
 HOSTCC	= clang
 HOSTCXX	= clang++
-else
-HOSTCC	= gcc
-HOSTCXX	= g++
-endif
+#else
+#HOSTCC	= gcc
+#HOSTCXX	= g++
+#endif
 KBUILD_HOSTCFLAGS   := -Wall -Wmissing-prototypes -Wstrict-prototypes -O2 \
 		-fomit-frame-pointer -std=gnu89 $(HOST_LFS_CFLAGS) \
 		$(HOSTCFLAGS)
@@ -410,7 +410,7 @@ KBUILD_HOSTLDLIBS   := $(HOST_LFS_LIBS) $(HOSTLDLIBS)
 
 # Make variables (CC, etc...)
 CPP		= $(CC) -E
-ifneq ($(LLVM),)
+#ifneq ($(LLVM),)
 CC		= clang
 LD		= ld.lld
 AR		= llvm-ar
@@ -420,17 +420,17 @@ OBJDUMP		= llvm-objdump
 READELF		= llvm-readelf
 OBJSIZE		= llvm-size
 STRIP		= llvm-strip
-else
-CC		= $(CROSS_COMPILE)gcc
-LD		= $(CROSS_COMPILE)ld
-AR		= $(CROSS_COMPILE)ar
-NM		= $(CROSS_COMPILE)nm
-OBJCOPY		= $(CROSS_COMPILE)objcopy
-OBJDUMP		= $(CROSS_COMPILE)objdump
-READELF		= $(CROSS_COMPILE)readelf
-OBJSIZE		= $(CROSS_COMPILE)size
-STRIP		= $(CROSS_COMPILE)strip
-endif
+#else
+#CC		= $(CROSS_COMPILE)gcc
+#LD		= $(CROSS_COMPILE)ld
+#AR		= $(CROSS_COMPILE)ar
+#NM		= $(CROSS_COMPILE)nm
+#OBJCOPY		= $(CROSS_COMPILE)objcopy
+#OBJDUMP		= $(CROSS_COMPILE)objdump
+#READELF		= $(CROSS_COMPILE)readelf
+#OBJSIZE		= $(CROSS_COMPILE)size
+#STRIP		= $(CROSS_COMPILE)strip
+#endif
 PAHOLE		= pahole
 LEX		= flex
 YACC		= bison
diff --git a/kernel/gen_kheaders.sh b/kernel/gen_kheaders.sh
index c1510f0ab3ea..1f1daafabc77 100755
--- a/kernel/gen_kheaders.sh
+++ b/kernel/gen_kheaders.sh
@@ -85,10 +85,17 @@ find $cpio_dir -type f -print0 |
 # Create archive and try to normalize metadata for reproducibility.
 # For compatibility with older versions of tar, files are fed to tar
 # pre-sorted, as --sort=name might not be available.
+
+#tar present in prebuilts doesnot support -I option. So using tar from host
+#bin_tar="/bin/tar"
+#find $cpio_dir -printf "./%P\n" | LC_ALL=C sort | \
+#	$bin_tar "${KBUILD_BUILD_TIMESTAMP:+--mtime=$KBUILD_BUILD_TIMESTAMP}" \
+#    --owner=0 --group=0 --numeric-owner --no-recursion \
+#    -I $XZ -cf $tarfile -C $cpio_dir/ -T - > /dev/null
 find $cpio_dir -printf "./%P\n" | LC_ALL=C sort | \
-    tar "${KBUILD_BUILD_TIMESTAMP:+--mtime=$KBUILD_BUILD_TIMESTAMP}" \
+	tar "${KBUILD_BUILD_TIMESTAMP:+--mtime=$KBUILD_BUILD_TIMESTAMP}" \
     --owner=0 --group=0 --numeric-owner --no-recursion \
-    -I $XZ -cf $tarfile -C $cpio_dir/ -T - > /dev/null
+    -J -cf $tarfile -C $cpio_dir/ -T - > /dev/null
 
 echo $headers_md5 > kernel/kheaders.md5
 echo "$this_file_md5" >> kernel/kheaders.md5
-- 
2.17.1

